app/controllers/users/sessions_controller.rb

```diff
@@ -24,4 +32,59 @@ class Users::SessionsController < Devise::SessionsController
   # def configure_sign_in_params
   #   devise_parameter_sanitizer.permit(:sign_in, keys: [:attribute])
   # end
+
+  include ApplicationHelper
+  def authenticate_with_otp_two_factor
```

Comment:
@ajaystha42 great job in adding a concern for this. Can you roll in the lines [here](https://github.com/restarone/violet_rails/pull/1153/files#diff-513a92c45a3a5c14210895bd4082fb0d9b6a33dea5209ff25362f03d2a1f300bR37-R89) into the concern?

Reply:
@donrestarone  Sure